rts/Game/GameControllerTextInput.cpp

```diff
@@ -384,45 +384,45 @@ bool GameControllerTextInput::CheckHandlePasteCommand(const std::string& rawLine
 }
 
 
-bool GameControllerTextInput::ProcessKeyPressAction(int key, const Action& action) {
+bool GameControllerTextInput::ProcessKeyPressAction(int keyCode, int scanCode, const Action& action) {
 	assert(userWriting);
 
 	if (action.command == "pastetext")
-		return (HandlePasteCommand(key, action.rawline));
+		return (HandlePasteCommand(keyCode, action.rawline));
 
 	if (action.command.find("edit_") == 0)
-		return (HandleEditCommand(key, action.command));
+		return (HandleEditCommand(keyCode, scanCode, action.command));
 
 	if (action.command.find("chatswitch") == 0)
-		return (HandleChatCommand(key, action.command));
+		return (HandleChatCommand(keyCode, action.command));
 
 	return false;
 }
 
 
-bool GameControllerTextInput::ConsumePressedKey(int key, const std::vector<Action>& actions) {
+bool GameControllerTextInput::ConsumePressedKey(int keyCode, int scanCode, const std::vector<Action>& actions) {
 	if (!userWriting)
 		return false;
 
 	for (const Action& action: actions) {
-		if (!ProcessKeyPressAction(key, action))
+		if (!ProcessKeyPressAction(keyCode, scanCode, action))
 			continue;
 
 		// the key was used, ignore it (ex: alt+a)
-		ignoreNextChar = keyCodes.IsPrintable(key);
+		ignoreNextChar = keyCodes.IsPrintable(keyCode);
 		break;
 	}
 
 	return true;
 }
 
 
-bool GameControllerTextInput::ConsumeReleasedKey(int key) const {
+bool GameControllerTextInput::ConsumeReleasedKey(int keyCode, int scanCode) const {
```

Comment:
Any reason why you are passing the scancode here if not using it?

Reply:
Probably for the same reason you requested Action to receive it, to propagate them to possible consumers. I could not do that but it would feel a bit silly to only have keycodes on the first layer of key(down|up) handling. I don't mind reducing the surface area of propagation, as long as future engine developers are aware they can retrieve this information.